### PR TITLE
Bump golang to 1.11.3 (CVE-2018-16875)

### LINKED ISSUE #72032: Update to go1.11.4
The go team has released 1.11.3 with three security fixes.

From their e-mail announcement:
```
cmd/go: remote command execution during "go get -u"
The issue is CVE-2018-16873 and Go issue golang.org/issue/29230. See the Go issue for details.
Thanks to Etienne Stalmans from the Heroku platform security team for discovering and reporting this issue.

cmd/go: directory traversal in "go get" via curly braces in import paths
The issue is CVE-2018-16874 and Go issue golang.org/issue/29231. See the Go issue for details.
Thanks to ztz of Tencent Security Platform for discovering and reporting this issue.

crypto/x509: CPU denial of service in chain validation
The issue is CVE-2018-16875 and Go issue golang.org/issue/29233. See the Go issue for details.
Thanks to Netflix for discovering and reporting this issue.
```

We should kick off upgrades of both the kubernetes build system, and our CI, as soon as possible.

@jayconrod: Do you have an ETA for the rules_go bump that will contain 1.11.3 toolchain support?

---
cc: @ixdy @kubernetes/sig-testing @kubernetes/sig-release @kubernetes/product-security-team

### PR DESCRIPTION
**Pull request by seemethere**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

In the same vein as https://github.com/kubernetes/kubernetes/pull/70665

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/72032

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Update to use go1.11.3 with fix for CVE-2018-16875
```

### CODE REVIEW

**Comment by cblecker** [comment]
@seemethere Has the Docker Hub tag for golang:1.11.3 been pushed yet?

/assign cblecker ixdy 
/uncc jbeda

**Comment by cblecker** [comment]
We also need to bump the rules_go version. The new tags are in the referenced issue.

**Comment by seemethere** [comment]
Yup images for amd64 we're pushed around 3pm PST

**Comment by ixdy** [comment]
yeah, can you please bump rules_go (in the `WORKSPACE` file) to 0.16.4?

**Comment by seemethere** [comment]
@cblecker @ixdy just pushed with `rules_go` updated to `0.16.4`

File: build/root/WORKSPACE [hunk 1]
```diff
@@ -5,7 +5,7 @@ load("//build:workspace.bzl", "CNI_VERSION", "CRI_TOOLS_VERSION")
 http_archive(
     name = "io_bazel_rules_go",
     sha256 = "f87fa87475ea107b3c69196f39c82b7bbf58fe27c62a338684c20ca17d1d8613",
```
Comment (cblecker) [inline build/root/WORKSPACE]: This also needs to be updated with the new hash of the new file.

**Review by cblecker — CHANGES_REQUESTED** [review]

**Comment by ixdy** [comment]
also `test/images/Makefile`?

we definitely specify the Go version in too many places :\

/ok-to-test

**Comment by ixdy** [comment]
I'm working on building the crossbuild image now.

File: build/root/WORKSPACE [hunk 2]
```diff
@@ -4,8 +4,8 @@ load("//build:workspace.bzl", "CNI_VERSION", "CRI_TOOLS_VERSION")
 
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "f87fa87475ea107b3c69196f39c82b7bbf58fe27c62a338684c20ca17d1d8613",
-    urls = mirror("https://github.com/bazelbuild/rules_go/releases/download/0.16.2/rules_go-0.16.2.tar.gz"),
+    sha256 = "62ec3496a00445889a843062de9930c228b770218c735eca89c67949cd967c3f"
```
Comment (ixdy) [inline build/root/WORKSPACE]: oh, a comma was dropped here

Comment (seemethere) [reply build/root/WORKSPACE]: re-added the comma

**Comment by ixdy** [comment]
The cross-build image is failing to build:
```
```Step 7/12 : RUN echo "deb http://archive.ubuntu.com/ubuntu xenial main universe" > /etc/apt/sources.list.d/cgocrosscompiling.list   && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 40976EAF437D05B5 3B4FE6ACC0B21F32   && apt-get update   && apt-get install -y build-essential   && for platform in ${KUBE_DYNAMIC_CROSSPLATFORMS}; do apt-get install -y crossbuild-essential-${platform}; done   && apt-get clean && rm -rf /var/lib/apt/lists/*
 ---> Running in 540f2d08fdbf
Warning: apt-key output should not be parsed (stdout is not a terminal)
Executing: /tmp/apt-key-gpghome.WdHS7PRkq5/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys 40976EAF437D05B5 3B4FE6ACC0B21F32
gpg: cannot open '/dev/tty': No such device or address
The command '/bin/sh -c echo "deb http://archive.ubuntu.com/ubuntu xenial main universe" > /etc/apt/sources.list.d/cgocrosscompiling.list   && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 40976EAF437D05B5 3B4FE6ACC0B21F32   && apt-get update   && apt-get install -y build-essential   && for platform in ${KUBE_DYNAMIC_CROSSPLATFORMS}; do apt-get install -y crossbuild-essential-${platform}; done   && apt-get clean && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 2
Makefile:24: recipe for target 'build' failed```
```

I'm not sure what is wrong. I can't build `cross-build:v1.11.2-1` anymore either, so I don't think it's related to this change.

**Comment by ixdy** [comment]
aha, just found https://stackoverflow.com/questions/53371626/apt-key-command-works-on-shell-but-fails-on-dockerfile.

Can you add `--no-tty` to the `apt-key` command in the Dockerfile?

basically this:
```patch
diff --git a/build/build-image/cross/Dockerfile b/build/build-image/cross/Dockerfile
index 8e4bc0f884..70e42cf87e 100644
--- a/build/build-image/cross/Dockerfile
+++ b/build/build-image/cross/Dockerfile
@@ -44,7 +44,7 @@ RUN apt-get update \
 # Use dynamic cgo linking for architectures other than amd64 for the server platforms
 # To install crossbuild essential for other architectures add the following repository.
 RUN echo "deb http://archive.ubuntu.com/ubuntu xenial main universe" > /etc/apt/sources.list.d/cgocrosscompiling.list \
-  && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 40976EAF437D05B5 3B4FE6ACC0B21F32 \
+  && apt-key adv --no-tty --keyserver keyserver.ubuntu.com --recv-keys 40976EAF437D05B5 3B4FE6ACC0B21F32 \
   && apt-get update \
   && apt-get install -y build-essential \
   && for platform in ${KUBE_DYNAMIC_CROSSPLATFORMS}; do apt-get install -y crossbuild-essential-${platform}; done \
```

**Review by cblecker — APPROVED** [review]

**Pull request merged**